lib/pg.provider.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const coreErrors = require('./errors');
+const { StorageProvider } = require('./provider');
+const {
+  generateQueryParams,
+  escapeIdentifier,
+  buildWhere,
+} = require('./pg.utils');
+const { PostgresCursor } = require('./pg.cursor');
+
+const metaschema = require('metaschema');
+const pg = require('pg');
+
+const path = require('path');
+
+const errors = {
+  objNotFound: id => new Error(
+    `No object with Id ${id} available`
+  ),
+  invalidSchema: errors => new Error(
+    `Invalid schema provided: ${errors.join('; ')}`
+  ),
+};
+
+class PostgresProvider extends StorageProvider {
+  // Create PostgresProvider
+  //   gs - globalstorage instance
+  constructor(gs) {
+    super(gs);
+
+    this.pool = null;
+  }
+
+  // Open PostgresProvider
+  //   options - object, passed to pg
+  //   callback - function(err, StorageProvider)
+  open(options, callback) {
+    metaschema.load(path.join(__dirname, '../schemas/system'), err => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      this.pool = new pg.Pool(options);
+      callback(null, this);
+    });
+  }
+
+  // Close PostgresProvider
+  //   callback - function(err)
+  close(callback) {
+    if (!this.pool) {
+      throw new Error('close() called before open()');
+    }
+    this.pool.end(() => {
+      this.pool = null;
+      callback();
+    });
+  }
+
+  // Generate globally unique id
+  //   callback - function(err, id)
+  // eslint-disable-next-line no-unused-vars
+  takeId(callback) {
+    throw new Error(coreErrors.NOT_IMPLEMENTED);
+  }
+
+  getCategoryById(id, callback) {
+    const categoryQuery = 'SELECT "Category" FROM "Identifier" WHERE "Id" = $1';
+    console.log('categoryQuery', categoryQuery);
+    this.pool.query(categoryQuery, [id], (err, res) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      if (res.rowCount === 0) {
+        callback(errors.objNotFound(id));
+        return;
+      }
+      const { Category } = res.rows[0];
+      callback(null, Category);
+    });
+  }
+
+  // Get object from GlobalStorage
+  //   id - globally unique object id
+  //   callback - function(err, obj)
+  get(id, callback) {
+    this.getCategoryById(id, (err, category) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      const objectQuery = `SELECT * FROM ${category} WHERE "Id" = $1`;
```

Comment:
`limit 1`?

Reply:
`"Id"` has a unique constraint on it already, so I don't think this will help in any way.